File: app/src/main/java/com/allever/spycamera/utils/ImageUitl.java
package com.allever.spycamera.utils;

import android.graphics.Bitmap;
import android.graphics.Matrix;

/**
 * Created by Allever on 18/5/11.
 */

public class ImageUitl {
    /**
     * 获取旋转后的图片
     * @param b
     * @param rotateDegree 旋转角度
     * @return
     */
    public static Bitmap getRotateBitmap(Bitmap b, float rotateDegree){
        if (b == null){
            return null;
        }
        Matrix matrix = new Matrix();
        matrix.postRotate((float)rotateDegree);
        Bitmap rotaBitmap = Bitmap.createBitmap(b, 0, 0, b.getWidth(), b.getHeight(), matrix, false);
        return rotaBitmap;
    }
}

File: app/src/main/java/com/allever/stealthcamera/utils/ImageUitl.kt
package com.allever.stealthcamera.utils

import android.graphics.Bitmap
import android.graphics.Matrix

/**
 * Created by Allever on 18/5/11.
 */

object ImageUitl {
    /**
     * 获取旋转后的图片
     * @param b
     * @param rotateDegree 旋转角度
     * @return
     */
    fun getRotateBitmap(b: Bitmap?, rotateDegree: Float): Bitmap? {
        if (b == null) {
            return null
        }
        val matrix = Matrix()
        matrix.postRotate(rotateDegree)
        return Bitmap.createBitmap(b, 0, 0, b.width, b.height, matrix, false)
    }
}
